public/catch-js-errors.js

```diff
@@ -108,8 +111,8 @@
 			}
 		}
 
-		if ( localStorage.getItem( 'log-errors' ) !== undefined &&
-			( localStorage.getItem( 'log-errors' ) === 'rest-api' || localStorage.getItem( 'log-errors' ) === 'true' )
+		if ( localStorageAssignment !== undefined &&
+			( localStorageAssignment === 'rest-api' || localStorageAssignment === 'true' )
```

Comment:
Maybe we could fix any localStorageAssignments that are set to 'true' before this if block.